tools/AWSFeedStorage.py

```diff
@@ -17,22 +19,63 @@ def __init__(self, uri):
 
         self.logger = logging.getLogger(__name__)
 
+        u = urlparse(uri)
+        self.s3_file = u.path[1:]
         self.s3_bucket = settings['AWS_S3_BUCKET']
-        self.s3_file = settings['AWS_S3_FILE_NAME']
-        self.s3 = boto3.resource('s3')
+        self.s3 = boto3.client('s3')
         self.dynamodb = DynamoDBConnector()
 
+    def _generate_datetag(self):
+        curdate = datetime.now()
+        return "{}".format(curdate.strftime('%Y%m%d'))
+
+    def _get_last_result(self):
+        try:
+            objs = self.s3.list_objects_v2(
+                Bucket=self.s3_bucket,
+                Prefix=self.s3_file
+            ).get('Contents', [])
+        except ClientError:
+            self.logger.warning('Could not connect to s3 bucket.')
+            return ''
+
+        if not objs:
+            self.logger.warning('Could not get last result file.')
+            return ''
+
+        last_added = [obj['Key'] for obj in sorted(
```

Comment:
as far as i understand it, this is quite inefficient in s3. is the idea that we will scrape periodically and always care about the last file scraped? can we not retrieve the date of last scrape somehow and recreate the key to fetch a file rather than always having to sort?

Reply:
I'm not happy with the way to store the results. We should think about it, but I'm pretty sure a file isn't the answer. I'm considering using Dynamo as a definitive storage, but I'm not sure about using a no-sql database yet. It comes with a lot of constraints. Maybe we should have a chat about it, as you will be pulling these data?

Reply:
happy to discuss whenever you have time (and i am not in a meeting :( )